### PR TITLE
fix theia-ide/theia #5513 Altered input box to remove readonly attribute

### PR DESCRIPTION
**Pull request by sauny**
Removed the readonly flag on the search input box

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
@sauny with your change I'm now able to re-enter text, but the search does not actually update.
It looks like there is more to do to actually fix #5513

**Comment by vince-fugnitto** [comment]
If you'd like to test, this is my following use case:
1. open the `explorer` view
2. start typing, initiating the search
3. open a file
4. return to the explorer view, setting the focus on the search box which is not readonly anymore
5. type new content to search
6. the search does not take effect (nodes are not highlighted)

**Comment by sauny** [comment]
@vince-fugnitto my apologies... all fixed now (I hope)

File: packages/core/src/browser/tree/search-box.ts [hunk 1]
```diff
@@ -171,8 +172,9 @@ export class SearchBox extends BaseWidget {
         this.addClass(SearchBox.Styles.SEARCH_BOX);
 
         const input = document.createElement('input');
-        input.readOnly = true;
+        input.readOnly = false;
         input.type = 'text';
+        input.onkeydown = (ev) => this.handle.bind(this)(ev);
```
Comment (vince-fugnitto) [inline packages/core/src/browser/tree/search-box.ts]: You can write this as:

```ts
input.onkeydown = ev => this.handle.bind(this)(ev);
```

It is the reason for the build failing https://travis-ci.org/theia-ide/theia/jobs/549920261#L861

**Comment by sauny** [comment]
Tried to sort this on the move. Will fix when I get back to my laptop!

**Comment by vince-fugnitto** [comment]
@kittaakos I hadn't seen your comments in the issue, is this something you don't want?
I think it adds value to be able to update the input directly and re-search, a lot like how the search-in-file is. If you disagree, perhaps we need to revert?

**Pull request merged**